homeassistant/components/modbus.py

```diff
@@ -33,20 +40,20 @@
     vol.Required(CONF_STOPBITS): vol.Any(1, 2),
     vol.Required(CONF_TYPE): 'serial',
     vol.Optional(CONF_TIMEOUT, default=3): cv.socket_timeout,
-}
+})
 
-ETHERNET_SCHEMA = {
+ETHERNET_SCHEMA = BASE_SCHEMA.extend({
     vol.Required(CONF_HOST): cv.string,
     vol.Required(CONF_PORT): cv.port,
     vol.Required(CONF_TYPE): vol.Any('tcp', 'udp', 'rtuovertcp'),
     vol.Optional(CONF_TIMEOUT, default=3): cv.socket_timeout,
-}
-
+})
 
 CONFIG_SCHEMA = vol.Schema({
-    DOMAIN: vol.Any(SERIAL_SCHEMA, ETHERNET_SCHEMA)
-}, extra=vol.ALLOW_EXTRA)
-
+    DOMAIN:
+        vol.All(cv.ensure_list,
+                vol.Schema([vol.Any(SERIAL_SCHEMA, ETHERNET_SCHEMA)]))
```

Comment:
Do we need schema definition here? Try without it.

Reply:
No, I just think we don't need to add `vol.Schema` around the list of schemas. Have you tried without that?

Reply:
Sorry, got confused. I removed it and that seems to work.